crates/voicevox_core/src/engine/kana_parser.rs

```diff
@@ -0,0 +1,280 @@
+}
+
+#[allow(dead_code)] // TODO: remove this feature
+fn create_kana(accent_phrases: &[AccentPhraseModel]) -> String {
+    let mut text = String::new();
+    for phrase in accent_phrases {
+        let moras = &phrase.moras;
+        for (index, mora) in moras.iter().enumerate() {
+            if ["A", "E", "I", "O", "U"].contains(&mora.vowel.as_ref()) {
+                text.push(UNVOICE_SYMBOL);
+            }
+            text.push_str(&mora.text);
+            if index + 1 == phrase.accent {
+                text.push(ACCENT_SYMBOL);
+            }
+        }
+        if phrase.is_interrogative {
+            text.push(WIDE_INTERROGATION_MARK);
+        }
+        text.push(if phrase.pause_mora.is_some() {
+            PAUSE_DELIMITER
+        } else {
+            NOPAUSE_DELIMITER
+        });
+    }
+    text.pop(); // remove last delimiter
+    text
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::engine::mora_list::MORA_LIST_MINIMUM;
+    use pretty_assertions::assert_eq;
+    use rstest::rstest;
+
+    #[rstest]
+    #[case(Some("da"), "ダ")]
+    #[case(Some("N"), "ン")]
+    #[case(Some("cl"), "ッ")]
+    #[case(Some("sho"), "ショ")]
+    #[case(Some("u"), "ウ")]
+    #[case(Some("gA"), "_ガ")]
+    #[case(Some("byO"), "_ビョ")]
+    #[case(Some("O"), "_オ")]
+    #[case(None, "fail")]
+    fn test_text2mora_with_unvoice(#[case] mora: Option<&str>, #[case] text: &str) {
+        let text2mora = &TEXT2MORA_WITH_UNVOICE;
+        assert_eq!(text2mora.len(), MORA_LIST_MINIMUM.len() * 2 - 2); // added twice except ン and ッ
+        let res = text2mora.get(text);
+        assert_eq!(mora.is_some(), res.is_some());
+        if let Some(res) = res {
+            let mut m = String::new();
+            if let Some(ref c) = res.consonant {
+                m.push_str(c);
+            }
+            m.push_str(&res.vowel);
+            assert_eq!(m, mora.unwrap());
+        }
+    }
+
+    #[test]
+    fn test_text_to_accent_phrase() {
+        let text_ok = ["ア_シタ'ワ", "ユウヒガ'", "_キ'レイ"];
+        let text_err = [
+            "アクセントナシ",
+            "アクセ'ント'タクサン'",
+            "'アクセントハジマリ",
+            "不明な'文字",
+        ];
+        for text in text_ok {
+            let result = text_to_accent_phrase(text);
+            assert!(result.is_ok(), "{:?}", result);
+        }
+        for text in text_err {
+            let result = text_to_accent_phrase(text);
+            assert!(result.is_err(), "{:?}", result);
+        }
+    }
+
+    #[test]
+    fn test_parse_kana() {
+        let text_ok = ["テ'ス_ト/テ_ス'ト、_テ'_スト？/テ'ス_ト？"];
+        let text_err = [
+            "クウハクノ'//フレーズ'",
+            "フレー？ズノ'/トチュウニ'、ギモ'ンフ",
+        ];
+        for text in text_ok {
+            let result = parse_kana(text);
+            assert!(result.is_ok(), "{:?}", result);
+        }
+        for text in text_err {
+            let result = parse_kana(text);
+            assert!(result.is_err(), "{:?}", result);
+        }
+    }
+    #[test]
```

Comment:
このテストは入力と出力の組みが１組しかないので `#[case]` は書かなくてよいと思いますが、一応 rstest にするのが良いと思います。

```suggestion
    #[rstest]
```